pymatgen/io/lobster/inputs.py

```diff
@@ -116,176 +115,140 @@ class Lobsterin(UserDict, MSONable):
-        Similar to the diff in INCAR.
+        Diff function for Lobsterin. Compare two Lobsterin and find
+        which parameters are the same. Similar to the diff method of Incar.
 
         Args:
-            other (Lobsterin): Lobsterin object to compare to
+            other (Lobsterin): Lobsterin object to compare to.
 
         Returns:
-            dict with differences and similarities
+            dict: {"Same": same_params, "Different": diff_params}
         """
-        similar_param = {}
-        different_param = {}
-        key_list_others = [element.lower() for element in other]
+        same_params = {}
+        diff_params = {}
 
+        # Check self
         for k1, v1 in self.items():
-            k1_lower = k1.lower()
-            k1_in_other = next((key_here for key_here in other if key_here.lower() == k1_lower), k1_lower)
-            if k1_lower not in key_list_others:
-                different_param[k1.lower()] = {"lobsterin1": v1, "lobsterin2": None}
+            if k1 not in other:
+                diff_params[k1] = {"lobsterin1": v1, "lobsterin2": None}
+
+            # String keywords
             elif isinstance(v1, str):
-                if v1.strip().lower() != other[k1_lower].strip().lower():
-                    different_param[k1.lower()] = {
-                        "lobsterin1": v1,
-                        "lobsterin2": other[k1_in_other],
-                    }
+                if v1 != other[k1]:
+                    diff_params[k1] = {"lobsterin1": v1, "lobsterin2": other[k1]}
                 else:
-                    similar_param[k1.lower()] = v1
+                    same_params[k1] = v1
+
+            # List keywords
             elif isinstance(v1, list):
                 new_set1 = {element.strip().lower() for element in v1}
-                new_set2 = {element.strip().lower() for element in other[k1_in_other]}
+                new_set2 = {element.strip().lower() for element in other[k1]}
                 if new_set1 != new_set2:
-                    different_param[k1.lower()] = {
-                        "lobsterin1": v1,
-                        "lobsterin2": other[k1_in_other],
-                    }
-            elif v1 != other[k1_lower]:
-                different_param[k1.lower()] = {
-                    "lobsterin1": v1,
-                    "lobsterin2": other[k1_in_other],
-                }
+                    diff_params[k1] = {"lobsterin1": v1, "lobsterin2": other[k1]}
+
+            # Float/boolean keywords
+            elif v1 != other[k1]:
+                diff_params[k1] = {"lobsterin1": v1, "lobsterin2": other[k1]}
             else:
-                similar_param[k1.lower()] = v1
+                same_params[k1] = v1
 
+        # Check other
         for k2, v2 in other.items():
-            if (
-                k2.lower() not in similar_param
-                and k2.lower() not in different_param
-                and k2.lower() not in [key.lower() for key in self]
```

Comment:
This check logic looks a bit over-complicated. I hope I didn't misunderstand here.

If `k2` is not in `self`, it then cannot be in the union `same_params`.

Reply:
I really went recently through this code and it all worked. I am very much afraid so many drastic chances might break a lot of things. This will take a lot of review time that I currently don't really have. We might need to leave this for a few weeks.

Reply:
(sounds harsher than intended 🙈. I just hoped I could have a quick look 😅. In any case, thank you for the woek)

Reply:
All good :), there isn't as much change as it seems to be to the code itself (sorry I'm making quite some changes to the comments which might have flooded the difference).

The core change is the case handling surround `UserDict/dict` functionalities. Now it would store the `Lobsterin` keywords(keys) internally as lowered as we don't need to lower every time we try to `get` a key or such.